### DISCUSSION TITLE
Caching of composer build if nothing has changed between deployments

### DISCUSSION THREAD

**Discussion by jamesmacwhite** [craftcms/cloud#44] [Feature Requests & Enhancements]
I don't know how feasible this is given the serverless design, but being hosted on similar ephemeral hosting environments like Fortrabbit, using an external CI like Buddy, we could cache the composer deployment step if the composer.lock hadn't changed between deployments. and only when the composer.lock was changed, would the composer build be rebuilt This reduced the deployment time when there were no composer changes, given the composer build was just pulled from the cache on these runs.

The average the build time of the project we are testing is around 5-8 minutes, reviewing the completed build logs, on average 40 seconds is down to the composer build step.

This might not be possible given Fortrabbit isn't serverless and you have an SSH deploy service, which you don't have with Craft Cloud but just opening a discussion as an area for consideration or thoughts.

**Comment by timkelty** [comment]
@jamesmacwhite we've already been exploring this, along with other improvements to deployment speed. We will keep you updated!

**Reply by jamesmacwhite** [reply]
Just had a build just complete in sub 5 minutes under the new BETA build/deployment process which is some serious performance improvement from the existing builder where it seems to range from 8 - 30 minutes and quite wide ranging most of the time.

If that is consistent under the new build process, take a bow, because that's a serious improvement!

**Reply by timkelty** [reply]
@jamesmacwhite all credit due to @deleugpn 🥇 

FWIW, while a caching solution may not be as feasible, we're still aiming for future improvements to get the build time down even more (e.g. parallellizing node/composer).

**Reply by jamesmacwhite** [reply]
@deleugpn Solid work! The sub-5 min builds are great. If parallelism with node and composer is something that happens, we could see maybe 2-3 min builds, which for serverless is amazing.

**Reply by deleugpn** [reply]
thanks @jamesmacwhite! I'm glad you're enjoying it!! I'm taking a peak at some other areas right now, but hopefully we can deliver some more improvements to build time soon
